### PR TITLE
Add updating shadow frustum on changing molecule

### PR DESCRIPTION
**Pull request by Hadegda**
## Description

Add updating shadow camera position and frustum on changing molecule place and on load/unload molecules

## Type of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read [CONTRIBUTING](https://github.com/epam/miew/blob/master/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/epam/miew/blob/master/CODE_OF_CONDUCT.md) guides.
- [x] I have followed the code style of this project.
- [x] I have run `yarn run ci`: lint and tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation / The changes do not need docs update.

### CODE REVIEW

File: src/Miew.js [hunk 1]
```diff
@@ -210,6 +210,31 @@ function _setContainerContents(container, element) {
   parent.appendChild(element);
 }
 
+Miew.prototype._updateShadowCamera = function () {
+  this._gfx.scene.updateMatrixWorld();
+  for (let i = 0; i < this._gfx.scene.children.length; i++) {
+    if (this._gfx.scene.children[i].type === 'DirectionalLight') {
+      const light = this._gfx.scene.children[i];
+      const shadowMatrix = new THREE.Matrix4().copy(light.shadow.matrix);
```
Comment (NataliaDSmirnova) [inline src/Miew.js]: At first please be very careful with on-the-fly object creation. The operation is expensive, so I ask you to reuse the once created variable as a temporary object to reduce using of `new` operator

File: src/Miew.js [hunk 2]
```diff
@@ -210,6 +210,31 @@ function _setContainerContents(container, element) {
...
+      const bBox = this._getbBox(shadowMatrix);
+
+      const direction = new THREE.Vector3().subVectors(light.target.position, light.position);
```
Comment (NataliaDSmirnova) [inline src/Miew.js]: Here is the same issue.

File: src/Miew.js [hunk 3]
```diff
@@ -889,6 +919,40 @@ Miew.prototype._getBSphereRadius = function () {
   return radius * this._objectControls.getScale();
 };
 
+Miew.prototype._getbBox = function (matrix) {
+  // calculate bounding box that would include all visuals and being axis aligned in world defined by
+  // transformation matrix: matrix
+  const OBB = new THREE.Box3();
```
Comment (NataliaDSmirnova) [inline src/Miew.js]: Please check the code also.

File: src/Miew.js [hunk 4]
```diff
@@ -889,6 +919,40 @@ Miew.prototype._getBSphereRadius = function () {
   return radius * this._objectControls.getScale();
 };
 
+Miew.prototype._getbBox = function (matrix) {
```
Comment (NataliaDSmirnova) [inline src/Miew.js]: Here I have got some questions for the function name. 
I think it should be _getOBB, also we should mention in comments that this is OBB for the whole scene consisted of several molecules and current usage of that OBB.

File: src/Miew.js (hunk 1 above)
Comment (Hadegda) [reply src/Miew.js]: Thanks for your comment
Understood the reason and fixed this code

**Pull request merged**